index.bs

```diff
@@ -1129,13 +1138,68 @@ a numbered step. If outdented, it (today) is rendered as a bullet in the midst o
 
                   1. If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{authenticatorAttachment}}</code> is
                     [=present|present=] and its value is not equal to |authenticator|'s [=authenticator attachment modality=], [=iteration/continue=].
-                  1. If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{requireResidentKey}}</code> is set to
-                    [TRUE] and the |authenticator| is not capable of storing a [=client-side-resident public key credential source=],
-                    [=iteration/continue=].
+
+                  1. If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{residentKey}}</code>
+
+                      <dl class="switch">
+                          :   is set to {{ResidentKeyRequirement/required}}
+                          ::  If the |authenticator| is not capable of storing a [=client-side-resident public key credential
+                              source=], [=iteration/continue=].
+
+                          :   is set to {{ResidentKeyRequirement/preferred}}
+                          ::  If the |authenticator| is not capable of storing a [=client-side-resident public key credential
```

Comment:
I don't think this is the behaviour that we would actually implement. Rather, `preferred` would mean that a resident credential would be created if possible, but if the user selects an authenticator incapable of that, we're not going to tell them to use the other one but rather will create a non-resident credential.

Reply:
+1. That's what I inferred.

Reply:
I have changed the algorithm to say that if the value is preferred or discouraged, then "no effect" - i.e. don't ignore authenticators and continue in this case, but instead let the value drive how we instruct the authenticator to work in the next part of the algorithm.